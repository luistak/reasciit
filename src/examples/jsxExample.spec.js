import Reasciit from './../../build';
import jsxTemplate from './jsxExample';

const order = {
	id: '1234',
	restaurantName: 'My cool restaurant with big name that occupy two lines',
	paymentMethod: 'MONEY',
	items: [{
		name: 'Big Hamburger with huge amount of cheese',
		quantity: 2,
		totalPrice: 46.9,
		observations: 'Please can you remove the onion on bottom.',
		subItems: [{
			name: 'Some extra cheese and sauce',
			quantity: 1,
			totalPrice: 5.1
		}, {
			name: 'Fries',
			quantity: 2,
			totalPrice: 2.1
		}]
	}, {
		name: 'Ice cream',
		quantity: 3,
		totalPrice: 13.05,
		observations: 'More chocolate.'
	}],
	deliveryFee: 6.05,
	address: {
		street: 'My street with a very very long name',
		number: '999',
		state: 'AC',
		city: 'My city',
		neighborhood: 'My neighborhood'
	}
};


describe('Invoice example', () => {
	it('should match jsx based on object', () => {
		const rendered = Reasciit.renderToString(jsxTemplate(order), new Reasciit(40));
		expect(rendered).toMatchSnapshot();
	});
});
