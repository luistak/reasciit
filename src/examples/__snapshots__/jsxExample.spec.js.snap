// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Invoice example should match jsx based on object 1`] = `
"----------------------------------------
               Order: 1234              
                                        
Restaurant: My cool restaurant with big 
name that occupy two lines              
Client address: My street with a very   
very long name, 999 - My neighborhood,  
My city - AC                            
Cidade: My city - AC                    
                                        
Items                                   
Qtt  Description                   Price
2    Big Hamburger with huge      $ 46.9
     amount of cheese                   
     1 Some extra cheese and       $ 5.1
     sauce                              
     2 Fries                       $ 2.1
                                        
3    Ice cream                   $ 13.05
                                        
         Parcial:                $ 59.95
         Delivery fee:            $ 6.05
         -------------------------------
         Total:                     $ 66
                                        
Payment method                          
MONEY                                   
                                        
                                        "
`;
